File: kq-projection-dotnet/kq-projection/Common/Models/Events/ProjectionEventResponse.cs
namespace KQ_Projection.Common.Models.Events;

using KQ_Projection.Common.Enums;
using System.Text.Json.Serialization;

public class ProjectionEventResponse
{
    [JsonPropertyName("playerId")]
    public long playerId { get; set; }

    [JsonPropertyName("gameId")]
    public Guid gameId { get; set; }

    [JsonPropertyName("eventStreamId")]
    public int? eventStreamId { get; set; }

    [JsonPropertyName("revertFromEventStreamId")]
    public int? revertFromEventStreamId { get; set; }

    [JsonPropertyName("eventType")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public EventType eventType { get; set; }

    [JsonPropertyName("initiatedBy")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public EventInitiatedBy initiatedBy { get; set; }

    [JsonPropertyName("timestamp")]
    public long timestamp { get; set; }

    [JsonPropertyName("responseMessage")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public ResponseMessage responseMessage { get; set; }

    [JsonPropertyName("responseData")]
    public ResponseData responseData { get; set; }
}

File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/ProjectionEventResponse.java
package com.kafkaquest.kq.common.util.models.events;

import com.kafkaquest.kq.common.util.models.events.responsedata.ResponseData;
import com.kafkaquest.kq.common.util.models.events.enums.EventInitiatedBy;
import com.kafkaquest.kq.common.util.models.events.enums.EventType;
import com.kafkaquest.kq.common.util.models.events.enums.ResponseMessage;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProjectionEventResponse {

    private Long playerId;
    private UUID gameId;
    private Integer eventStreamId;
    private Integer revertFromEventStreamId;
    private EventType eventType;
    private EventInitiatedBy initiatedBy;
    private long timestamp;
    private ResponseMessage responseMessage;
    private ResponseData responseData;

}
